src/app/home-module/services/student-service.service.ts

```diff
@@ -0,0 +1,83 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { Student } from '../models/student';
+import { environment } from 'src/environments/environments';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class StudentServiceService {
+  BaseUrl = environment.BaseUrL;
+
+  constructor(private http: HttpClient) {}
+
+  public ageCalculation(birthday: Date): number {
+    const today = new Date();
+    const birthDate = new Date(birthday);
+    let age = today.getFullYear() - birthDate.getFullYear();
+    const m = today.getMonth() - birthDate.getMonth();
+    if (m < 0 || (m === 0 && today.getDate() < birthDate.getDate())) {
+      age--;
+    }
+
+    return age;
+  }
+
+  public getStudents(): Observable<Student[]> {
+    return new Observable<Student[]>(observer => {
+      this.http
+        .get<Student[]>(this.BaseUrl)
+        .subscribe((students: Student[]) => {
+          observer.next(students);
+        });
+    });
+  }
+
+  public addStudent(student: Student): Observable<Student> {
+    const postObj = {
```

Comment:
move the object creation where add student function being called.in that way you function responsible only for the one task which send the post request. And make the above change here as well.